### ISSUE TITLE
core: Meditate on the attribute names situation.

### ISSUE THREAD

**Issue by PapyChacal** [xdslproject/xdsl#2719]
As brought up in:
- #2718 
We currently annotate the base `Attribute` class with a name `ClassVar`. I.e., it is expected  that *any* subclass defines a name - including further attribute base classes, used as interfaces for example.
The only practical issue with this I've observed is in IRDL verifiers, that actually uses this transparently. Then, using such a base class errors if it has no name.
More on the principles side of things, I feel they should all define a name if we so say in the base class indeed. So we could:
- Not expect *all* of them to have a name anymore, but only at the `irdl_attr_definition` stage
- Or just enforce indeed that they all have names.

**Comment by superlopuh** [comment]
My take on this is that we don't want inheritance in IRDL, if we can avoid it, so the approach in #2718, with an override of a class variable, is not ideal. This is separate to our Python implementation, where inheritance seems like the most ergonomic approach. What I'm trying to say is that it would be great to use this task to track a discussion of how to define abstract attributes and use them in constraints, and also how to map these to PyRDL.

**Comment by alexarice** [comment]
Have we done enough meditating here? I feel the name situation is ok.

**Comment by superlopuh** [comment]
Yeah me too, and the issue feels not very actionable to me anyway, I think we can close this.
